### PR TITLE
Turn promise tests into sequential promise tests.

### PR DESCRIPTION
**Pull request by g-ortuno**
Same idea as #96 but using promise_test:

`The goal [is] to make tests that share state (across the tests, or in the object under test) be deterministic.`

How should I test this change doesn't break stuff?

### CODE REVIEW

**Comment by g-ortuno** [comment]
@jgraham please review?

File: testharness.js [hunk 1]
```diff
@@ -1583,6 +1585,7 @@ policies and contribution forms [3].
     function Tests()
     {
         this.tests = [];
+        this.promise_tests = Promise.resolve();
```
Comment (scheib) [inline testharness.js]: It should be clear to a reader why this exists, and that the intent is that all calls to promise_test will have the tests run sequentially. Either with a comment or renaming the member to e.g.:
promise_test_sequence
sequential_promise_tests
promise_test_chain

**Comment by scheib** [comment]
docs/api.md should also be updated to document that tests created with promise_test will be executed sequentially. And, as this is being done because of issues found with async_test running tests in parallel, there should be a clear warning added to async_test documentation of that limitation.

File: testharness.js (hunk 1 above)
Comment (g-ortuno) [inline testharness.js]: Done.

**Comment by g-ortuno** [comment]
Done.

**Comment by g-ortuno** [comment]
@jgraham Is there anyway I can get this merged in? There are a couple of places in chromium where people are coding their own sequential promise tests ([here](https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/LayoutTests/http/tests/fetch/resources/fetch-test-helpers.js&sq=package:chromium&l=18) and [here](https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/LayoutTests/bluetooth/bluetooth-helpers.js&sq=package:chromium)). This could be avoided if we modify the promise test or make a new sequential_promise_test. Any thoughts?

**Comment by jgraham** [comment]
See the open issues on critic

**Comment by jgraham** [comment]
Sorry, dropped the ball here, but I replied to the comments on critic again (add your email address!)

**Comment by jgraham** [comment]
(more comments)

**Comment by g-ortuno** [comment]
PR squashed. Ready for merge.

**Pull request merged**